src/dotty/tools/dotc/core/pickling/TreePickler.scala

```diff
@@ -531,8 +535,16 @@ class TreePickler(pickler: TastyPickler) {
       withLength { pickleType(ann.symbol.typeRef); pickleTree(ann.tree) }
     }
 
+    def updateMapWithDeltas[T](mp: collection.mutable.Map[T, Addr]) = {
+      mp.map{
+        case (key, addr) => (key, adjusted(addr))
```

Comment:
Any reason to avoid `AnyRefMap`? IMO that should be the first choice when you need a map in the compiler. Its main claim to fame is that it can calls `equals/hashCode` rather than `==`/``##` which is a non-negligible performance win.